tests/functionnal/Knowbase.php

```diff
@@ -0,0 +1,231 @@
+         'a_attr' => ['data-id' => '0']
+      ];
+
+      $tree = \Knowbase::getJstreeCategoryList();
+      $this->array($tree)->isEqualTo([$expected_root_cat]);
+
+      // Add a private item (not FAQ)
+      $kbitem = new \KnowbaseItem();
+      $kbitem_id = $kbitem->add(
+         [
+            'knowbaseitemcategories_id' => $cat_1_1_2_id,
+            'users_id' => \Session::getLoginUserID(),
+         ]
+      );
+      $this->integer($kbitem_id)->isGreaterThan(0);
+
+      $kbitem_target = new \Entity_KnowbaseItem();
+      $kbitem_target_id = $kbitem_target->add(
+         [
+            'knowbaseitems_id' => $kbitem_id,
+            'entities_id' => 0,
+            'is_recursive' => 1,
+         ]
+      );
+      $this->integer($kbitem_target_id)->isGreaterThan(0);
+
+      // Check that tree contains root + category branch containing kb item of user
+      $tree = \Knowbase::getJstreeCategoryList();
+      $this->array($tree)->isEqualTo(
+         [
+            [
+               'id' => "$cat_1_1_2_id",
+               'parent' => "$cat_1_1_id",
+               'text' => 'cat 1.1.2 <strong title="This category contains articles">(1)</strong>',
+               'a_attr' => ['data-id' => "$cat_1_1_2_id"]
+            ],
+            [
+               'id' => "$cat_1_1_id",
+               'parent' => "$cat_1_id",
+               'text' => 'cat 1.1',
+               'a_attr' => ['data-id' => "$cat_1_1_id"]
+            ],
+            [
+               'id' => "$cat_1_id",
+               'parent' => '0',
+               'text' => 'cat 1',
+               'a_attr' => ['data-id' => "$cat_1_id"]
+            ],
+            $expected_root_cat
+         ]
+      );
+
+      // Add a FAQ item
+      $kbitem = new \KnowbaseItem();
+      $kbitem_id = $kbitem->add(
+         [
+            'knowbaseitemcategories_id' => $cat_1_2_1_id,
+            'is_faq' => 1,
+         ]
+      );
+      $this->integer($kbitem_id)->isGreaterThan(0);
+
+      $kbitem_target = new \Entity_KnowbaseItem();
+      $kbitem_target_id = $kbitem_target->add(
+         [
+            'knowbaseitems_id' => $kbitem_id,
+            'entities_id' => 0,
+            'is_recursive' => 1,
+         ]
+      );
+      $this->integer($kbitem_target_id)->isGreaterThan(0);
+
+      $this->setUp(); // Force session reset
+
+      // Expected root category item for anonymous user
+      $expected_root_cat = [
+         'id' => '0',
+         'parent' => '#',
+         'text' => 'Root category',
+         'a_attr' => ['data-id' => '0']
+      ];
+
+      // Check that tree contains root only (FAQ is not public) for anonymous user
+      $tree = \Knowbase::getJstreeCategoryList();
+      $this->array($tree)->isEqualTo([$expected_root_cat]);
+
+      // Check that tree contains root + category branch containing FAQ item (FAQ is public) for anonymous user
+      global $CFG_GLPI;
+      $CFG_GLPI['use_public_faq'] = 1;
```

Comment:
If you change a global value, you'd better reset it (and before tests can fail; after `$tree =` I guess here)

Reply:
Done here https://github.com/glpi-project/glpi/pull/5508/files#diff-3dddccf6317f001ae18582ee376c1f0fR235

Reply:
sounds good